### PR TITLE
Update reported tx count to exclude errors

### PR DESCRIPTION
**Pull request by sagar-solana**
#### Problem

All attempted txs were counting towards tps. This included txs that errored out. 

#### Summary of Changes

Updated counter to only include txs that can be recorded.

### CODE REVIEW

**Comment by garious** [comment]
As you make changes like this, please update this document that details how each metric is calculated: https://github.com/solana-labs/solana/blob/master/book/src/performance-metrics.md

File: core/src/bank_forks.rs [hunk 1]
```diff
@@ -110,12 +110,19 @@ impl BankForks {
...
+            .map(|bank| bank.transaction_count())
+            .unwrap_or(0);
         root_bank.squash();
+        let new_tx_count = root_bank.transaction_count();
         self.prune_non_root(root);
         inc_new_counter_info!(
             "bank-forks_set_root_ms",
             timing::duration_as_ms(&set_root_start.elapsed()) as usize
         );
+        inc_new_counter_info!("bank-forks_set_root_tx_count", new_tx_count - root_tx_count);
```
Comment (sagar-solana) [inline core/src/bank_forks.rs]: @aeyakovenko, this counter will track squashed tx count. I haven't added it to the dashboard in this pr.

**Comment by sakridge** [comment]
Can we merge this?

**Comment by sagar-solana** [comment]
Pending a perf testnet run. Was hoping @pgarg66 could help but he was OoO yesterday and I am today.

**Comment by pgarg66** [comment]
@danpaul000 FYI
we also need to update https://github.com/solana-labs/solana/blob/master/ci/testnet-automation.sh

File: metrics/testnet-monitor.json [hunk 2]
```diff
@@ -502,7 +502,7 @@
           "hide": false,
           "orderByTime": "ASC",
           "policy": "default",
-          "query": "SELECT ROUND(MEAN(\"sum\")) FROM ( SELECT sum(\"count\") FROM \"$testnet\".\"autogen\".\"counter-banking_stage-process_transactions\" WHERE $timeFilter GROUP BY time(1s) )\n\n",
+          "query": "SELECT ROUND(MEAN(\"sum\")) FROM ( SELECT sum(\"count\") FROM \"$testnet\".\"autogen\".\"counter-banking_stage-record_transactions\" WHERE $timeFilter GROUP BY time(1s) )\n\n",
```
Comment (pgarg66) [inline metrics/testnet-monitor.json]: A similar change is needed in mean, max and total transactions counts in the dashboard

image

**Pull request merged**